### PR TITLE
Feat/sales mode, table and menu --run-build

### PR DESCRIPTION
**Pull request by jokomyn**
changelog:
- feat sales mode
- feat menu
- feat quick service and table
- enhance code

### CODE REVIEW

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: NONE

failed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs:  [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities:  [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells:  [0 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage: **Coverage**

- Duplications: **Duplication**

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells: security_a [1 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=3)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=3)

File: src/app/components/Features/QuickService/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import * as QuickService from './QuickService.component';
```
Comment (akarintitech) [inline src/app/components/Features/QuickService/index.ts:1]: **Remove this unused import of 'QuickService'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells: security_a [1 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=3)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=3)

File: src/app/components/Features/QuickService/index.ts (hunk 1 above)
Comment (akarintitech) [inline src/app/components/Features/QuickService/index.ts:1]: **Remove this unused import of 'QuickService'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells: security_a [10 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=3)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=3)

File: src/app/modules/order/redux/OrderRedux.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import {Action} from '@reduxjs/toolkit'
+import {persistReducer} from 'redux-persist'
+import storage from 'redux-persist/lib/storage'
+import {call, put, takeLatest} from 'redux-saga/effects'
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:4]: **Remove this unused import of 'call'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:4]: **Remove this unused import of 'put'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:4]: **Remove this unused import of 'takeLatest'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

File: src/app/modules/order/redux/OrderRedux.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import {ENV_STORAGE_KEY, env} from 'src/app/utils/env-utils'
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:5]: **Remove this unused import of 'ENV_STORAGE_KEY'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

File: src/app/modules/order/redux/OrderRedux.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import {getProfile} from '../../permissions/services/Permission.services'
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:6]: **Remove this unused import of 'getProfile'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

File: src/app/modules/order/redux/OrderRedux.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import { OrderModel } from '../models/Order.model'
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:7]: **Remove this unused import of 'OrderModel'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

File: src/app/modules/order/redux/OrderRedux.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  export interface ActionWithPayload<T> extends Action {
+    payload?: T
+  }
+
+  export interface IOrderState {
+    status?: FetchingStatusState
+    isBookAndOrder?: boolean
+    numberPax?: number|null|string
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:22]: **Replace this union type with a type alias.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S4323&rule_key=typescript:S4323) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *5min* effort

File: src/app/modules/order/redux/OrderRedux.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        payload: {menuItem}
+    }),
+    RemoveMenu: (menuItem:Array<Item>) =>({
+        type: actionTypes.RemoveMenu,
+        payload: {menuItem}
+    })
+
+  }
+
+//   export function* saga() {
```
Comment (akarintitech) [inline src/app/modules/order/redux/OrderRedux.ts:99]: **Remove this commented out code.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S125&rule_key=typescript:S125) 

CODE_SMELL Codesmell　　MAJOR **Major**　　:hourglass: *5min* effort

File: src/app/utils/cek-json-utils.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export const checkJson = (data:string) =>{
+    if (/^[\],:{}\s]*$/.test(data.replace(/\\["\\\/bfnrtu]/g, '@').
+        replace(/"[^"\\\n\r]*"|true|false|null|-?\d+(?:\.\d*)?(?:[eE][+\-]?\d+)?/g, ']').
```
Comment (akarintitech) [inline src/app/utils/cek-json-utils.ts:3]: **Simplify this regular expression to reduce its complexity from 26 to the 20 allowed.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S5843&rule_key=typescript:S5843) 

CODE_SMELL Codesmell　　MAJOR **Major**　　:hourglass: *20min* effort

File: src/app/components/Features/QuickService/index.ts (hunk 1 above)
Comment (akarintitech) [inline src/app/components/Features/QuickService/index.ts:1]: **Remove this unused import of 'QuickService'.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S1128&rule_key=typescript:S1128) 

CODE_SMELL Codesmell　　MINOR **Minor**　　:hourglass: *2min* effort

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells: security_a [1 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)

- Closed Issues: 9 issues closed

## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=3)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=3)

File: src/app/utils/cek-json-utils.ts (hunk 8 above)
Comment (akarintitech) [inline src/app/utils/cek-json-utils.ts:3]: **Simplify this regular expression to reduce its complexity from 26 to the 20 allowed.**  [<sub>Why is this an issue?</sub>](https://sonar.akarinti.tech/coding_rules?open=typescript:S5843&rule_key=typescript:S5843) 

CODE_SMELL Codesmell　　MAJOR **Major**　　:hourglass: *20min* effort

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=3)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=3)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=3)

- Code Smells: security_a [0 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=3)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=3&inNewCodePeriod=true)

- Closed Issues: 10 issues closed

## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=3)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=3)

**Review by aldiz1711 — APPROVED** [review]
OK

**Pull request merged**